src/isar/models/mission/mission.py

```diff
@@ -56,18 +53,29 @@ def is_finished(self) -> bool:
                 # Not all steps have been completed yet
                 return False
 
-        # Check if the task has been marked as partially successful by having one or
-        # more inspection steps fail
+        return True
+
+    def update_task_status(self) -> None:
+        for step in self.steps:
+            if step.status is StepStatus.Failed and isinstance(step, MotionStep):
+                self.status = TaskStatus.Failed
+                return
+
+            elif (step.status is StepStatus.Failed) and isinstance(
+                step, InspectionStep
+            ):
+                self.status = TaskStatus.PartiallySuccessful
+                continue
+
+            elif step.status is StepStatus.Successful:
+                continue
```

Comment:
I agree with this functionality, but it's a bit hard to read, maybe we could change it to be "queries" on the array like
if any(blabla in self.steps) ?